trustmanager/x509utils.go

```diff
@@ -507,18 +505,18 @@ func X509PublicKeyID(certPubKey data.PublicKey) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	var finalAlgorithm data.KeyAlgorithm
+	pubKeyBytes, err := x509.MarshalPKIXPublicKey(cert.PublicKey)
+	if err != nil {
+		return "", err
+	}
 
+	var key data.PublicKey
 	switch certPubKey.Algorithm() {
 	case data.ECDSAx509Key:
-		finalAlgorithm = data.ECDSAKey
+		key = data.NewECDSAPublicKey(pubKeyBytes)
```

Comment:
Yeah, it's another existing problem. If a keystore gets corrupted, it would be possible for us to identify a private key with the wrong ID given that the IDs are based on the public component only. If we wanted to be really careful, we could validate everything we sign with the associated public key, but if we're doing things right, that level of paranoia shouldn't be necessary.